src/cli/mod.rs

```diff
@@ -1,15 +1,32 @@
 mod config;
-use crate::mainnet::Client;
+use crate::{mainnet::Client, Error};
 pub use config::CliConfig;
 use config::{BeaconMethod, Namespace::Beacon};
+use std::fmt;
 
-pub async fn run_cli(client: &Client, args: &CliConfig) {
+pub async fn run_cli(client: &Client, args: &CliConfig) -> Result<Box<dyn fmt::Debug>, Error> {
     match &args.namespace {
         Beacon(BeaconMethod::Genesis) => {
-            println!("{:?}", &client.get_genesis_details().await.unwrap());
+            let result = client.get_genesis_details().await?;
+            Ok(Box::new(result))
         }
         Beacon(BeaconMethod::Root(arg)) => {
-            println!("{}", &client.get_state_root(arg.state_id.clone()).await.unwrap())
+            let result = client.get_state_root(arg.state_id.clone()).await?;
```

Comment:
if you rebase this from `main` you should be able to just pass by ref now

```suggestion
            let result = client.get_state_root(&arg.state_id).await?;
```